### ISSUE TITLE
Composer conflicts and missing files

### ISSUE THREAD

**Issue by vitoralmeidabr** [johnbillion/query-monitor#696]
Hello, John!

Thanks for your plugin, it's very helpfull daily.

But at this moment i dealing with some problems/conflicts betwen your composer dependencies versus my main project dependencies

Downloading your plugin from wordpress plugins or via composer, we have inside your package, the vendor path and inside we have two files who maps to an unknown file called "composer/InstalledVersions.php". Those files are:
> vendor/composer/autoload_static.php
>  ----   'Composer\\InstalledVersions' => \_\_DIR\_\_ . '/..' . '/composer/InstalledVersions.php',
> vendor/composer/autoload_classmap.php
>  ----   'Composer\\InstalledVersions' => $vendorDir . '/composer/InstalledVersions.php',

This file not exists in your package, it's generated through composer-runtime-api 2.0 (https://getcomposer.org/doc/07-runtime.md#installed-versions), wich thas not referenced in your composer.json

**What's happening?**
When in my code, i call "\Composer\InstalledVersions::getRootPackage()['version']" i got two warnings: 

> [03-Oct-2022 16:29:39 UTC] PHP Warning:  include(C:\apache\myproject.com\public\cms\wp-content\plugins\query-monitor\vendor\composer/../composer/InstalledVersions.php): failed to open stream: No such file or directory in C:\apache\myproject.com\vendor\composer\ClassLoader.php on line 571
> [03-Oct-2022 16:29:39 UTC] PHP Warning:  include(): Failed opening 'C:\apache\myproject.com\public\cms\wp-content\plugins\query-monitor\vendor\composer/../composer/InstalledVersions.php' for inclusion (include_path='.;C:/laragon/etc/php/pear') in C:\apache\myproject.com\vendor\composer\ClassLoader.php on line 571

**Why it's happening?**
If you add this file to package, either the conflicts will remain, because acttualy your plugin adds vendor path and ClassLoader class without prefixed vendor or your own namespace, and will override initial declaration of this class from my project.

**How handle that?**
Sugestion to handle that, please take a look to this plugin https://github.com/Yoast/wordpress-seo/ at vendor_prefixed and this lib : https://github.com/coenjacobs/mozart

> About
Developers tool for WordPress plugins: Wraps all your projects dependencies in your own namespace, in order to prevent conflicts with other plugins loading the same dependencies in different versions.

Expect I was clear and objective, but if you need more details or examples, tell me please.

Thanks in advanced for your attention and work!

**Comment by johnbillion** [comment]
Strange, the autoloader file in Query Monitor should only include classes from Query Monitor itself (therefore no need to use the `vendor_prefixed` method).

I'll take a look and see what the problem is. Thanks for the report.

**Comment by tomjn** [comment]
I'm encountering this too, we have 2 warnings for:

```
Failed opening '/usr/src/app/content/plugins-mu/vendor/query-monitor/vendor/composer/../composer/InstalledVersions.php' for inclusion
```

and:

```
vendor/composer/ClassLoader.php:571
Composer\Autoload\includeFile()
vendor/composer/ClassLoader.php:428
Composer\Autoload\ClassLoader->loadClass()
Unknown location
spl_autoload_call()
Unknown location
method_exists()
vendor/jean85/pretty-package-versions/src/PrettyVersions.php:46
Jean85\PrettyVersions::checkProvidedPackages()
vendor/jean85/pretty-package-versions/src/PrettyVersions.php:19
Jean85\PrettyVersions::getVersion()
vendor/sentry/sentry/src/ClientBuilder.php:107
Sentry\ClientBuilder->__construct()
vendor/sentry/sentry/src/ClientBuilder.php:115
Sentry\ClientBuilder::create()
vendor/sentry/sentry/src/functions.php:14
Sentry\init()
content/plugins-mu/loader.php:32
```

Of note the file it's looking for is only present when using Composer v2, if you're using v1 to distribute the .org plugin then this won't be a surprise

**Comment by tomjn** [comment]
Better stack trace:

<img width="1336" alt="Screenshot 2022-10-19 at 17 13 53" src="https://user-images.githubusercontent.com/58855/196746943-6af98d45-673d-4227-b0ae-464ed1928659.png">

**Comment by tomjn** [comment]
I changed into the folder Query monitor was installed to and ran `composer install --no-dev` with v2.4.3. The problem disappeared.

@johnbillion this is almost certainly a packaging problem related to the composer version installed

**Comment by tomjn** [comment]
I suspect the problem is that this github action doesn't update composer to v2 before dumping the autoloader:

https://github.com/johnbillion/query-monitor/blob/develop/.github/workflows/build.yml

**Comment by tomjn** [comment]
@johnbillion what's the plan/timeline to expect for 3.10.2?

**Comment by johnbillion** [comment]
This should also be fixed by 32d826b which sets the `prepend-autoloader` config to false, meaning the class map definitions for QM will have a lower priority than those from the main project and therefore `InstalledVersions` will load from the correct location.
